tests/data/test_minute_bars.py

```diff
@@ -451,6 +452,44 @@ def test_pad_data(self):
 
         self.assertEquals(100.0, volume_price)
 
+    def test_nans(self):
+        """
+        Test writing empty data.
+        """
+        sid = 1
+        last_date = self.writer.last_date_in_output_for_sid(sid)
+        self.assertIs(last_date, NaT)
+
+        self.writer.pad(sid, TEST_CALENDAR_START)
+
+        last_date = self.writer.last_date_in_output_for_sid(sid)
+        self.assertEqual(last_date, TEST_CALENDAR_START)
+
+        freq = self.market_opens.index.freq
+        minute = self.market_opens[TEST_CALENDAR_START + freq]
+        minutes = date_range(minute, minute + Timedelta('8 min'), freq='min')
+        data = DataFrame(
+            data={
+                'open': [nan, nan, nan, nan, nan, nan, nan, nan, nan],
```

Comment:
As far as I can tell `numpy` always uses the value `0b11111111111 << 51` (`\x00\x00\x00\x00\x00\x00\xf8\x7f`)
So I think it is safe to use the default nan representation value which numpy uses for these tests,
though may switch to using `full` instead of the manual array construction.

```
import sys
import numpy as np
from numpy import nan, full, uint32

print("Python version: {0}".format(sys.version))
print("numpy version: {0}".format(np.__version__))

nan_byte_string = b'\x00\x00\x00\x00\x00\x00\xf8\x7f'
manual_nan = np.int64(0b111111111111 << 51).view(np.float64)
assert nan_byte_string == manual_nan.tostring()


def nan_check(num):
    print("nan check for num {0}".format(num))
    nans = full(num, nan)
    print("array repr:")
    print(nans)
    as_byte_string = nans.tostring()
    print("byte string")
    print(repr(as_byte_string))
    assert as_byte_string == nan_byte_string * num
    print(nans.astype(uint32))

nan_check(9)
print("")
nan_check(10)


* Python 2, numpy 1.9.2

$ python nan_info.py
Python version: 2.7.10 (default, Sep 24 2015, 17:50:09)
[GCC 5.1.1 20150618 (Red Hat 5.1.1-4)]
numpy version: 1.9.2
nan check for num 9
array repr:
[ nan  nan  nan  nan  nan  nan  nan  nan  nan]
byte string
'\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f'
[0 0 0 0 0 0 0 0 0]

nan check for num 10
array repr:
[ nan  nan  nan  nan  nan  nan  nan  nan  nan  nan]
byte string
'\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f'
[2147483648 2147483648 2147483648 2147483648 2147483648 2147483648
 2147483648 2147483648          0          0]

* Python 3, numpy 1.9.2

$ python nan_info.py
Python version: 3.4.2 (default, Jul  9 2015, 17:24:30)
[GCC 5.1.1 20150618 (Red Hat 5.1.1-4)]
numpy version: 1.9.2
nan check for num 9
array repr:
[ nan  nan  nan  nan  nan  nan  nan  nan  nan]
byte string
b'\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f'
[2147483648 2147483648 2147483648 2147483648 2147483648 2147483648
 2147483648 2147483648          0]

nan check for num 10
array repr:
[ nan  nan  nan  nan  nan  nan  nan  nan  nan  nan]
byte string
b'\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f'
[2147483648 2147483648 2147483648 2147483648 2147483648 2147483648
 2147483648 2147483648          0          0]

* Python 2, numpy 1.10.4

$ python nan_info.py
Python version: 2.7.10 (default, Sep 24 2015, 17:50:09)
[GCC 5.1.1 20150618 (Red Hat 5.1.1-4)]
numpy version: 1.10.4
nan check for num 9
array repr:
[ nan  nan  nan  nan  nan  nan  nan  nan  nan]
byte string
'\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f'
[0 0 0 0 0 0 0 0 0]

nan check for num 10
array repr:
[ nan  nan  nan  nan  nan  nan  nan  nan  nan  nan]
byte string
'\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f\x00\x00\x00\x00\x00\x00\xf8\x7f'
[2147483648 2147483648 2147483648 2147483648 2147483648 2147483648
 2147483648 2147483648          0          0]
```